src/index/mod.rs

```diff
@@ -79,6 +79,9 @@ pub trait Index {
 
     /// Clear the entire index.
     fn reset(&mut self) -> Result<()>;
+
+    /// Remove data vectors represented by IDs.
+    fn remove_ids(&mut self, sel: &[faiss_sys::FaissIDSelector_H]) -> Result<()>;
```

Comment:
As discussed above, `sel` would become of type `&IdSelector`. Also, since the function provides `n_removed`, let's return that too: `-> Result<u64>`

Implementations would need to change accordingly.

Reply:
added `Result<i64>`, since that matches Faiss